### PR TITLE
Update compatibility tests for clients

### PR DESCRIPTION
**Pull request by NoyException**

### CODE REVIEW

**Comment by TianyuZhang1214** [comment]
It seems all tests for MySQL failed?

**Comment by fanyang01** [comment]
Great work! So many programming languages :-)

I guess these failures could be resolved by a simple fix. Could you please look into it?

File: compatibility/mysql/c/mysql_test.c [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                mysql_free_result(res);
+                return 0;
+            }
+            mysql_free_result(res);
+        }
+    }
+    return 1;
+}
+
+size_t removeNewline(char *line) {
```
Comment (TianyuZhang1214) [inline compatibility/mysql/c/mysql_test.c:87]: seems dumplicated: 
```c
size_t removeNewline(char *line) {
    size_t len = strlen(line);
    while (len > 0 && (line[len - 1] == '\n' || line[len - 1] == '\r')) {
        line[--len] = '\0';
    }
    return len;
}
```

**Review by TianyuZhang1214 — CHANGES_REQUESTED** [review]

File: compatibility/mysql/csharp/MySQLTest.cs [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    {
+        private MySqlConnection conn;
+        private MySqlCommand cmd;
+        private List<Test> tests = new List<Test>();
+
+        public void Connect(string ip, int port, string user, string password)
+        {
+            try
+            {
+                string connectionString = $"Server={ip};Port={port};User Id={user};Password={password};";
```
Comment (TianyuZhang1214) [inline compatibility/mysql/csharp/MySQLTest.cs:19]: This line should be moved in front of the try-catch block:
```cs
string connectionString = $"Server={ip};Port={port};User Id={user};Password={password};";
```

File: compatibility/mysql/csharp/MySQLTest.cs [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                conn = new MySqlConnection(connectionString);
+                conn.Open();
+                cmd = conn.CreateCommand();
+                cmd.CommandType = CommandType.Text;
+            }
+            catch (MySqlException e)
+            {
+                throw new Exception(e.Message);
```
Comment (TianyuZhang1214) [inline compatibility/mysql/csharp/MySQLTest.cs:27]: Please add necessary prefix for all `Exception`. For example, the `Exception` here can be: 
```cs
                throw new Exception($"Failed to connect to MySQL server: {ex.Message}", ex);

```

File: compatibility/mysql/csharp/MySQLTest.cs [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+            public bool Run(MySqlCommand cmd)
+            {
+                try
+                {
+                    Console.WriteLine("Running test: " + query);
+                    cmd.CommandText = query;
+                    using (var reader = cmd.ExecuteReader())
+                    {
+                        if (!reader.HasRows)
```
Comment (TianyuZhang1214) [inline compatibility/mysql/csharp/MySQLTest.cs:108]: It's better to move result validation logic to a separate method `ValidateResults`.

**Review by TianyuZhang1214 — CHANGES_REQUESTED** [review]

**Comment by fanyang01** [comment]
Why does the Postgres compatiability test fail for more languages? Previously, it only failed for Java.

File: compatibility/mysql/go/mysql.go [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+	scanner := bufio.NewScanner(file)
+	var query string
+	var results [][]string
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if line == "" {
+			if query != "" {
+				t.addTest(query, results)
+				query = ""
```
Comment (TianyuZhang1214) [inline compatibility/mysql/go/mysql.go:64]:
```go
query, results = "", nil
```

File: compatibility/mysql/go/mysql.go [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+		fmt.Println("Invalid port:", os.Args[2])
+		os.Exit(1)
+	}
+	user := os.Args[3]
+	password := os.Args[4]
+	testFile := os.Args[5]
+
+	tests := &Tests{}
+	tests.connect(ip, port, user, password)
+	tests.readTestsFromFile(testFile)
```
Comment (TianyuZhang1214) [inline compatibility/mysql/go/mysql.go:172]:
```go
    defer tests.disconnect()

	tests.readTestsFromFile(testFile)

	if !tests.runTests() {
		os.Exit(1)
	}
```

File: compatibility/mysql/go/mysql.go [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+			rowPointers[i] = &row[i]
+		}
+		if err := rows.Scan(rowPointers...); err != nil {
+			fmt.Println("Error scanning row:", err)
+			return false
+		}
+
+		for i, expected := range test.expectedResults[rowCount] {
+			if row[i] != expected {
+				fmt.Printf("Expected: '%s', got: '%s'\n", expected, row[i])
```
Comment (TianyuZhang1214) [inline compatibility/mysql/go/mysql.go:138]: Print the row id here: 
```go
fmt.Printf("Mismatch at row %d, column %d: expected '%s', got '%s'\n", rowCount+1, i+1, expected, row[i])
```

**Review by TianyuZhang1214 — CHANGES_REQUESTED** [review]

**Comment by fanyang01** [comment]
Let's comment out failed languages to make the CI health and fix the failures one-by-one in the next step.

File: compatibility/mysql/csharp/MySQLTest.cs (hunk 4 above)
Comment (TianyuZhang1214) [reply compatibility/mysql/csharp/MySQLTest.cs:108]: Got it, let's keep this one as it is.

**Pull request merged**